Add Roadmaps as a way to encourage contributions file

Riaankl opened the pull request:
This is the output of a google docs collaberation.
All comments in the doc was resolved and closed.

fixes: #329

Riaankl commented:
@TheFoxAtWork @dzolotusky 
This is a band new document, can you please review with you TOC hats on.

TheFoxAtWork commented on website/content/maintainers/community/contributor-growth-framework/open-source-roadmaps.md:
```diff
@@ -0,0 +1,127 @@
+
+# Importance of Roadmaps in Open Source Projects
+
+A roadmap is a strategic plan that outlines the goals and direction of an open source project. For open source projects, having a well-defined roadmap is crucial.
```
Could we include characteristics of a roadmap or more specifically what the benefit of strategic planning is to projects?  Some projects may not use the traditional GitHub implementation of a "roadmap" so I want to ensure we are outcome oriented and not prescriptive of the tool/mechanism to achieve that outcome.

Riaankl replied:
Added a section on "Roadmaps bring a stratigic advantage"
Maybe I missed the point at 2am in the morning. :smile:

TheFoxAtWork replied:
I don't think I was clear in my comment, apologies. I meant that strategic planning is important for projects to perform and roadmaps are an essential tool to capture that planning so adopters, contributors, and others understand where a project is headed, when it plans to get there, and how it will do it.

Riaankl replied:
I added a bullet point for "strategic planning"

TheFoxAtWork commented on website/content/maintainers/community/contributor-growth-framework/open-source-roadmaps.md:
```diff
@@ -0,0 +1,127 @@
+A roadmap is a strategic plan that outlines the goals and direction of an open source project. For open source projects, having a well-defined roadmap is crucial.
+
+- A roadmap provides a **clear vision and purpose for the project**, helping contributors understand the project's objectives and direction. This clarity motivates them to work towards common goals.
+- Roadmaps **aid project management** by setting priorities, allocating resources, and coordinating efforts. They ensure that development tasks are well-organized and yield meaningful outcomes.
```
```suggestion
- Roadmaps may **aid project management**. They may be leveraged to set priorities, allocate resources, and coordinate efforts. When used effectively, they ensure that development tasks are well-organized and yield meaningful outcomes in a consistent, transparent manner.
```

TheFoxAtWork commented on website/content/maintainers/community/contributor-growth-framework/open-source-roadmaps.md:
```diff
@@ -0,0 +1,127 @@
+- A roadmap provides a **clear vision and purpose for the project**, helping contributors understand the project's objectives and direction. This clarity motivates them to work towards common goals.
+- Roadmaps **aid project management** by setting priorities, allocating resources, and coordinating efforts. They ensure that development tasks are well-organized and yield meaningful outcomes.
+- With many feature requests and contributions, roadmaps **help prioritize tasks** based on their importance and alignment with project goals. This keeps development efforts focused on key areas.
+- Transparently sharing the roadmap **attracts contributors** by showcasing ongoing efforts and how their skills fit into the project's future.
```
```suggestion
- Publishing the roadmap to the projects repo could **attract and retain contributors** by highlighting ongoing and planned efforts, enabling contributors to identify areas and times when their skills fit into the project's planned work.
```

TheFoxAtWork commented on website/content/maintainers/community/contributor-growth-framework/open-source-roadmaps.md:
```diff
@@ -0,0 +1,127 @@
+- Roadmaps **aid project management** by setting priorities, allocating resources, and coordinating efforts. They ensure that development tasks are well-organized and yield meaningful outcomes.
+- With many feature requests and contributions, roadmaps **help prioritize tasks** based on their importance and alignment with project goals. This keeps development efforts focused on key areas.
+- Transparently sharing the roadmap **attracts contributors** by showcasing ongoing efforts and how their skills fit into the project's future.
+- Roadmaps foster trust by **offering transparency into development plans and progress**, ensuring contributors feel confident in the project's management and also good to guide the users and downstream adoptions. Users and companies can receive advanced notifications about upcoming developments, enabling them to strategically devise their individualized utilization strategies for the software.
```
Couple items on this one:
- The first sentence doesn't read clearly, I'm not sure on what it is intending to convey.
- not all adopters or contributors are Companies, I would recommend using "Organizations" as it covers a variety of entities from business to government and non profit as well as considers sub-organizational elements for each of those that may operate with autonomy.

Riaankl replied:
- adopted the change to line 13
- Changed company to organization

TheFoxAtWork commented on website/content/maintainers/community/contributor-growth-framework/open-source-roadmaps.md:
```diff
@@ -0,0 +1,127 @@
+- With many feature requests and contributions, roadmaps **help prioritize tasks** based on their importance and alignment with project goals. This keeps development efforts focused on key areas.
+- Transparently sharing the roadmap **attracts contributors** by showcasing ongoing efforts and how their skills fit into the project's future.
+- Roadmaps foster trust by **offering transparency into development plans and progress**, ensuring contributors feel confident in the project's management and also good to guide the users and downstream adoptions. Users and companies can receive advanced notifications about upcoming developments, enabling them to strategically devise their individualized utilization strategies for the software.
+- Roadmaps **ensure development aligns with user demands**, enhancing the overall user experience.
```
```suggestion
- Roadmaps may allow the project to **align its development activities with adopter, user, and contributor feedback**, enhancing the overall experience for everyone.
```
I'm suggesting this rewrite as I think it is more favorable for collaboration between users and the projects, rather than projects in service to users. I know I'm guilty of using this phrasing in the past.

TheFoxAtWork commented on website/content/maintainers/community/contributor-growth-framework/open-source-roadmaps.md:
```diff
@@ -0,0 +1,127 @@
+- Roadmaps **ensure development aligns with user demands**, enhancing the overall user experience.
+- Roadmaps **help projects adapt to changes** in the open source landscape by incorporating feedback and addressing new challenges.
+- Roadmaps facilitate proactive identification and mitigation of potential challenges, **enabling the project to navigate risks effectively**.
+- Roadmaps guide the **efficient distribution of available resources**, preventing overextension of limited resources and ensuring tasks are adequately staffed and funded.
```
Do we have concerns with using "staffed and funded"? Most projects likely aren't hiring staff for these and funding is limited if present. I'm not sure of a more cohesive way of conveying intent however.

Riaankl replied:
"...ensuring tasks receive appropriate time and attention."
How about that?

TheFoxAtWork replied:
That reads much better and still gets the point across.

TheFoxAtWork commented on website/content/maintainers/community/contributor-growth-framework/open-source-roadmaps.md:
```diff
@@ -0,0 +1,127 @@
+- Roadmaps **help projects adapt to changes** in the open source landscape by incorporating feedback and addressing new challenges.
+- Roadmaps facilitate proactive identification and mitigation of potential challenges, **enabling the project to navigate risks effectively**.
+- Roadmaps guide the **efficient distribution of available resources**, preventing overextension of limited resources and ensuring tasks are adequately staffed and funded.
+- Roadmaps **ensure long-term sustainability** enabling the project to strategize for continuous growth and relevance, considering factors such as maintenance, community engagement, and evolving technology trends.
```
```suggestion
- Roadmaps **enable long-term sustainability** allow the project to strategize for continuous growth and evolution, considering factors such as maintenance, community engagement, and shifting technology trends.
```

TheFoxAtWork commented on website/content/maintainers/community/contributor-growth-framework/open-source-roadmaps.md:
```diff
@@ -0,0 +1,127 @@
+
+# Understanding Roadmaps in Open Source Projects
+
+A roadmap in an open source project is a strategic plan that outlines the project's vision, goals, and major milestones over a specific period. It serves as a navigational guide for project contributors, developers, and stakeholders, helping them understand the project's trajectory and how their efforts align with the larger objectives. Roadmaps come in various forms, tailored to meet the unique  ...
```
Recommend adding a line in here about multiple kinds of roadmaps maybe in use by projects for various needs.  They don't need limited to one.

Riaankl replied:
It's important to note that multiple kinds of roadmaps may be in use with in a single project to address various needs, ensuring flexibility and adaptability within the community.

How about this?

TheFoxAtWork replied:
perfect!

TheFoxAtWork reviewed (changes requested):
There is a lot of wonderful content here. I've made several suggestions and have open questions of a few items for enhanced clarity.

Riaankl commented:
> There is a lot of wonderful content here. I've made several suggestions and have open questions of a few items for enhanced clarity.

Thank you @TheFoxAtWork  for the thorough review. I believe I have addressed all of your review comments. Please let me know if any further updates are needed, or if I have misinterpreted anything.
